library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -302,15 +303,15 @@ def create(snapshot_type, description, previous: nil, cleanup: nil, important: f
         raise SnapperNotConfigured unless configured?
 
         cmd = format(CREATE_SNAPSHOT_CMD,
-          root:          target_root,
-          snapshot_type: snapshot_type,
-          description:   description)
-        cmd << " --pre-num #{previous.number}" if previous
+          root:          target_root.shellescape,
+          snapshot_type: snapshot_type.to_s.shellescape,
+          description:   description.shellescape)
+        cmd << " --pre-num #{previous.number.to_s.shellescape}" if previous
         cmd << " --userdata \"important=yes\"" if important
 
         if cleanup
           strategy = CLEANUP_STRATEGY[cleanup]
-          cmd << " --cleanup \"#{strategy}\"" if strategy
+          cmd << " --cleanup \"#{strategy.shellescape}\"" if strategy
```

Comment:
Bad idea: That string is already quoted within this command line; that's the `\"` before and after the strategy. If you shell-escape it in addition to that, this will be wrong.

Since the strategies are only a very limited set of fixed strings, this will most likely never cause problems, but it is contrary to the intent of this. Please remove this shellescape call.

Reply:
Explicit quotes with shellescape are plain wrong.

Reply:
OK, removing the quotes is also an option; but one way or the other, it needs to be consistent. Right now we have the embedded quotes with `\"` **plus** the shell-escape. That won't work.